### ISSUE TITLE
Wrong stack for errors

### ISSUE THREAD

**Issue by stalniy** [feathersjs/feathers-errors#78]
### Steps to reproduce
```js
/Users/stse/projects/mapapi/bla.js:5
throw new BadRequest('test')
^
Error
    at Object.<anonymous> (/Users/stse/projects/mapapi/node_modules/feathers-errors/lib/index.js:69:27)
    at Module._compile (module.js:569:30)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:503:32)
    at tryModuleLoad (module.js:466:12)
    at Function.Module._load (module.js:458:3)
    at Module.require (module.js:513:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/Users/stse/projects/mapapi/bla.js:1:86)
    at Module._compile (module.js:569:30)
stse:~/projects/mapapi$ cat bla.js
const { BadRequest } = require('feathers-errors')

console.log(123)

throw new BadRequest('test')
```

- [x] Tell us what broke. The more detailed the better.
- [x] If you can, please create a simple example that reproduces the issue and link to a gist, jsbin, repo, etc.

### Expected behavior
Stack should be captured at the moment of throwing errors. Please check this https://gist.github.com/justmoon/15511f92e5216fa2624b

### Actual behavior
Stack is captured at moment of requiring `feathers-errors` module

### System configuration

Tell us about the applicable parts of your setup.

**Module versions** (especially the part that's not working):

**NodeJS version**: 8.1.0

**Operating System**:

**Browser Version**:

**React Native Version**:

**Module Loader**:

**Comment by asabadyr** [comment]
Have same issue here. Calling a service which requires authorization without an Authorization header causes this stacktrace in console:
`< error:  Error
<     at Object.<anonymous> (/Users/iceraven/Projects/User Analytics/Backend/node_modules/feathers-errors/lib/index.js:69:27)
<     at Module._compile (module.js:571:32)
<     at Object.Module._extensions..js (module.js:580:10)
<     at Module.load (module.js:488:32)
<     at tryModuleLoad (module.js:447:12)
<     at Function.Module._load (module.js:439:3)
<     at Module.require (module.js:498:17)
<     at require (internal/module.js:20:19)
<     at Object.<anonymous> (/Users/iceraven/Projects/User Analytics/Backend/node_modules/feathers-rest/lib/wrappers.js:13:23)
<     at Module._compile (module.js:571:32)
`

client receives even more strange response:
`{
  "name": "GeneralError",
  "message": "`toJSON` takes 0 argument, 1 provided after:\nr.db(\"user_analytics\").table(\"fbusers\")",
  "code": 500,
  "className": "general-error",
  "data": {},
  "errors": {}
}`

**Comment by NikitaVlaznev** [comment]
Having the same issue.

**Comment by daffl** [comment]
I just tried the `captureStackTrace` from the Gist linked and it does not seem to do anything (as in after changing it there is no `stack` property anymore at all).

**Comment by stalniy** [comment]
@daffl which version of node or browser did you use?

**Comment by daffl** [comment]
Node v8.1.1. Anybody tried this with feathers-errors and got it working?

**Comment by stalniy** [comment]
I've just tested on node 8.1.3 and works as expected: https://nodejs.org/api/errors.html#errors_error_capturestacktrace_targetobject_constructoropt

**Comment by jonlambert** [comment]
I'm also experiencing this issue on Node v8.1.4. All errors returned are:

```
Error
    at Object.<anonymous> (/Users/Jon/Code/feathers/node_modules/feathers-errors/lib/index.js:69:27)
    at Module._compile (module.js:571:32)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)
    at Function.Module._load (module.js:439:3)
    at Module.require (module.js:498:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/Users/Jon/Code/feathers/node_modules/feathers-rest/lib/wrappers.js:13:23)
    at Module._compile (module.js:571:32)
```

I'm also using webpack to build the project - could this be contributing towards the problem?

**Comment by alaycock** [comment]
Also encountering this in Node v7.7.4. Getting the exact same stack trace as @jonlambert

**Comment by daffl** [comment]
I hopefully finally fixed this one in v2.9.0. It was pretty annoying that logging an error didn't give you any useful information.

Thanks for the info on captureStackTrace @stalniy.
